password_enhancements.module

```diff
@@ -0,0 +1,412 @@
+<?php
+
+/**
+ * @file
+ * Password related enchantments.
+ */
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\password_enhancements\Form\Settings;
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_theme().
+ */
+function password_enhancements_theme($existing, $type, $theme, $path): array {
+  return [
+    'password_enhancements_error_messages' => [
+      'variables' => [
+        'error_message' => '',
+        'messages' => [],
+      ],
+    ],
+    'password_enhancements_policy_constraints' => [
+      'variables' => [
+        'constraints' => [],
+        'descriptions' => [],
+        'hidden' => FALSE,
+      ],
+    ],
+    'password_enhancements_logged_out_message' => [
+      'variables' => [
+        'message' => '',
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function password_enhancements_form_user_form_alter(&$form, FormStateInterface $form_state, string $form_id): void {
+  password_enhancements_form_user_register_form_alter($form, $form_state, $form_id);
+  $form['password_change_required']['#access'] = Drupal::currentUser()->hasPermission('administer user password settings');
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function password_enhancements_form_password_enhancements_change_form_alter(&$form, FormStateInterface $form_state, $form_id): void {
+  password_enhancements_form_user_register_form_alter($form, $form_state, $form_id);
```

Comment:
Do not call a hook implementation directly. Create a dedicated function (marked as internal with `_`) that you can call everywhere it is needed.